common/src/main/java/com/netflix/conductor/common/metadata/tasks/Task.java

```diff
@@ -202,6 +202,9 @@ public boolean isRetriable() {
     @ProtoField(id = 42)
     private boolean subworkflowChanged;
 
+    @ProtoField(id = 43)
+    private long firstStartTime;
```

Comment:
should be added to `equals`, `hashcode` and `toString`?

Reply:
Actually, I think we should change the hashCode to be aware of the task's PK -- which is taskName. Thoughts @jmigueprieto

Reply:
@v1r3n  `hashCode` is considering `taskDefName`

```
  @Override
    public int hashCode() {
        return Objects.hash(
                getTaskType(),
                getStatus(),
                getInputData(),
                getReferenceTaskName(),
                getWorkflowPriority(),
                getRetryCount(),
                getSeq(),
                getCorrelationId(),
                getPollCount(),
                getTaskDefName(),
                ...
}
```

Or do you want to use just "task name"? 
**PROS**
- It will be simpler 
- (small) Performance gain

**CONS**
- We'll need to modify `equals`:  if only id is used in hashCode but equals considers additional fields, the contract will be broken. 
- We'll have an incomplete equality: this might break existing code e.g.: tests that assert `assertNotEquals(task0, task1)`. If both task have the same name but different fields, the assertion is `true` now but will be `false` is we just consider task name.

Reply:
I think we can analyze this further in another PR. But, for the moment I believe that we should stick to considering the added fields to `equals` and `hashCode`